### PR TITLE
Normalized Bounding Box update

### PR DESCRIPTION
**Pull request by I-Hong**
The PR is to update the detected bounding box and normalize it in the range between [0,1].

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
...
+This repository contains code to instantiate and deploy a facial age estimation model. The model detects faces in an image, extracts facial features for each face detected and finally predicts the age of each face. The model uses a coarse-to-fine strategy to perform multi-class classification and regression for age estimation. The input to the model is an image and the output is a list of estimated ages and bounding box coordinates of each face detected in the image. The format of the bounding box coordinates is `[xmin, ymin, width, height]`, where each coordinate is normalized by the height and width of the image dimension for y and x, respectively. Each coordinate is therefore in the range [0, 1].
```
Comment (ajbozarth) [inline README.md]: Just noticed that you forgot to update the coordinates def, it's it should now be `[ymin, xmin, ymax, xmax]`

File: core/model.py [hunk 2]
```diff
@@ -82,29 +82,27 @@ def _pre_process(self, input_img):
                 x1, y1, w, h = d['box']
                 x2 = x1 + w
                 y2 = y1 + h
+                # Normalized bbx coordinates
+                d['box'] = [float(x1) / img_w, float(y1) / img_h, float(x2) / img_w, float(y2) / img_h]
```
Comment (ajbozarth) [inline core/model.py]: Also based on my other previous delayed observation I noticed your coordinates are in a different order than the other models, it should be `[ymin, xmin, ymax, xmax]` (x and y switched). Though your's makes more sense the others are already like that.

Comment (ajbozarth) [reply core/model.py]: other examples to follow https://github.com/IBM/MAX-Facial-Emotion-Classifier https://github.com/IBM/MAX-Object-Detector

File: README.md (hunk 1 above)
Comment (I-Hong) [reply README.md]: Thanks @ajbozarth.

File: core/model.py (hunk 2 above)
Comment (I-Hong) [reply core/model.py]: Thanks @ajbozarth. Yea, I noticed that our versions are different to OpenCV, do you think it'd be better to consistent with OpenCV?

Comment (ajbozarth) [reply core/model.py]: I want to say we should, but I'm not sure making non-back compatible changes to multiple models is the best decision @MLnick what do you think

Comment (I-Hong) [reply core/model.py]: Yea, it'll make more things to change. Okay, I'll change the current version to fit ours. Thanks for the comments, @ajbozarth.

**Comment by I-Hong** [comment]
> Thanks for the updates, @MLnick should also aprove before we merge this though. Also once this is merged and #27 and #27 are addressed we should release a v3.0 version of the model. This release should be cut as soon as we can. Also once this is merged we will want to open issues and get fixes out for the web app and min I web app to support the changes.

Thanks @ajbozarth for the review and comments. Sure, once the PR is good to merge, I'll update web app part.

**Comment by I-Hong** [comment]
Hi @bdwyer2, do you have any idea about the error https://travis-ci.com/IBM/MAX-Facial-Age-Estimator/builds/109362642, i.e., the connection error?

**Comment by bdwyer2** [comment]
@I-Hong trying adding `numpy==1.16.2` to `requirements.txt`

File: core/util.py [hunk 3]
```diff
@@ -2,12 +2,12 @@
 
 # image resize if the size of width or height is larger than 1024
 def img_resize(input_data):
-    ratio = 1
+    # ratio = 1
```
Comment (ajbozarth) [inline core/util.py]: we should just remove this line if it's commented out

File: api/predict.py [hunk 4]
```diff
@@ -10,7 +10,8 @@
 
 label_prediction = MAX_API.model('LabelPrediction', {
     'age_estimation': fields.Integer(required=True, description='Estimated age for the face'),
-    'face_box': fields.List(fields.Integer(required=True), description='Bounding box coordinates for the face')
+    'detection_box': fields.List(fields.Float(required=True), description='Bounding box coordinates for the face, ' + \
+         'in the form of an array of normalized coordinates [xmin, ymin, xmax, ymax]. Each coordinate is in the range [0, 1]')
```
Comment (MLnick) [inline api/predict.py]: This is inconsistent with what is in the readme: https://github.com/IBM/MAX-Facial-Age-Estimator/pull/24/files#diff-04c6e90faac2675aa89e2176d2eec7d8R5

File: core/model.py [hunk 5]
```diff
@@ -71,7 +71,7 @@ def _pre_process(self, input_img):
         img_h, img_w, _ = np.shape(input_img)
 
         # check image w/h > 1024
-        input_img, ratio =img_resize(input_img)
+        input_img =img_resize(input_img)
```
Comment (MLnick) [inline core/model.py]: space after `=` here

File: tests/test.py [hunk 6]
```diff
@@ -34,10 +34,10 @@ def _check_response(r):
     json = r.json()
     assert json['status'] == "ok"
     assert 55 > json['predictions'][0]['age_estimation'] > 35
-    assert 575 > json['predictions'][0]['face_box'][0] > 560
-    assert 335 > json['predictions'][0]['face_box'][1] > 320
-    assert 720 > json['predictions'][0]['face_box'][2] > 700
-    assert 975 > json['predictions'][0]['face_box'][3] > 960
+    assert .32 > json['predictions'][0]['detection_box'][1] > .15
```
Comment (MLnick) [inline tests/test.py]: These ranges seem pretty wide - is it necessary to have it that wide?

File: api/predict.py (hunk 4 above)
Comment (I-Hong) [reply api/predict.py]: Thanks @MLnick

**Pull request merged**